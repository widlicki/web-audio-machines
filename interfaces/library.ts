import { Tier } from '.'

export interface AudioCategory {
  id: string
  name: string
  tier: Tier
}

export interface AudioFile {
  id: string
  name: string
  fileName: string
  category: string
  dateAdded: Date
  tags?: string[]
  tier: Tier
}
