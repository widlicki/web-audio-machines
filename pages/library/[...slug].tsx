import type { NextPage } from 'next'

const FilteredLibraryFile: NextPage = () => {
  return (
    <div>
      <h1>Filtered Library</h1>
    </div>
  )
}

export default FilteredLibraryFile
