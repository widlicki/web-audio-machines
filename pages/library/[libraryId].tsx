import type { NextPage } from 'next'

const LibraryFile: NextPage = () => {
  return (
    <div>
      <h1>Library File</h1>
    </div>
  )
}

export default LibraryFile
