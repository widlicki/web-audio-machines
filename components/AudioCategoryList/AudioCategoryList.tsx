import {
  Box,
  List,
  ListItem,
  ListItemButton,
  ListItemText,
} from '@mui/material'
import { useState } from 'react'
import { AudioCategory } from '../../interfaces/library'

export interface Props {
  categories: AudioCategory[]
}

const AudioCategoryList: React.FC<Props> = ({ categories }) => {
  const [selected, setSelected] = useState<string>('')

  const selectCat = (id: string) => {
    setSelected(id)
  }

  return (
    <Box>
      <List>
        {categories &&
          categories.map((cat) => (
            <ListItem key={cat.id}>
              <ListItemButton
                onClick={() => selectCat(cat.id)}
                selected={cat.id === selected}
              >
                <ListItemText primary={cat.name} />
              </ListItemButton>
            </ListItem>
          ))}
      </List>
    </Box>
  )
}

export default AudioCategoryList
