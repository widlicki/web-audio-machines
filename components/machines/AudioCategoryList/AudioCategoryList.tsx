import FolderIcon from '@mui/icons-material/Folder'
import FolderSpecialIcon from '@mui/icons-material/FolderSpecial'
import {
  Avatar,
  Box,
  List,
  ListItem,
  ListItemAvatar,
  ListItemButton,
  ListItemText,
} from '@mui/material'
import { useState } from 'react'
import { AudioCategory } from '../../../interfaces/library'
import { NotFoundMessage } from '../../../styles/Theme'

export interface AudioCategoryListProps {
  categories: AudioCategory[]
}

const AudioCategoryList = ({ categories }: AudioCategoryListProps) => {
  const [selected, setSelected] = useState<string>('')

  const selectCat = (id: string) => {
    setSelected(id)
  }

  return (
    <Box>
      <List>
        {categories && categories.length > 0 ? (
          categories.map((cat) => (
            <ListItem key={cat.id} sx={{ padding: 0 }}>
              <ListItemButton
                onClick={() => selectCat(cat.id)}
                selected={cat.id === selected}
              >
                <ListItemAvatar>
                  <Avatar>
                    {cat.tier === 'PREMIUM' ? (
                      <FolderSpecialIcon />
                    ) : (
                      <FolderIcon />
                    )}
                  </Avatar>
                </ListItemAvatar>
                <ListItemText primary={cat.name} />
              </ListItemButton>
            </ListItem>
          ))
        ) : (
          <NotFoundMessage>- No Data -</NotFoundMessage>
        )}
      </List>
    </Box>
  )
}

export default AudioCategoryList
