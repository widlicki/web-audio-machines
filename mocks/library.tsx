import { AudioCategory, AudioFile } from '../interfaces/library'

export const mockAudioCategories: AudioCategory[] = [
  {
    id: 'a0649ac4-9618-4dc0-993d-1dc4dcce1539',
    name: 'Kick',
    tier: 'FREE',
  },
  {
    id: 'feb9d1ce-5098-4a30-97bd-c248341615f4',
    name: 'Bass',
    tier: 'FREE',
  },
  {
    id: '998b77a6-25de-48c4-83ff-04e51cafeaaf',
    name: 'Synth',
    tier: 'PREMIUM',
  },
]

export const mockAudioFiles: AudioFile[] = [
  {
    id: '8f25cb68-7dd8-4335-9778-7950ea08a735',
    name: 'Kick-1',
    fileName: 'kick1.wav',
    category: 'Kick',
    dateAdded: new Date(2021, 12, 28),
    tags: ['kick', '808'],
    tier: 'FREE',
  },
  {
    id: 'c630edbe-59eb-4f90-ad08-b2448d3fa051',
    name: 'Kick-2',
    fileName: 'kick2.wav',
    category: 'Kick',
    dateAdded: new Date(2021, 12, 28),
    tags: ['kick', '909', 'house'],
    tier: 'FREE',
  },
  {
    id: '8a23c994-3c07-4276-ae48-531903d03005',
    name: 'Bass-1',
    fileName: 'bass1.wav',
    category: 'Bass',
    dateAdded: new Date(2021, 12, 28),
    tags: ['bass', 'minimal', 'techno'],
    tier: 'FREE',
  },
  {
    id: '7b506039-88c3-4f6a-8c64-75f89b9788f0',
    name: 'Synth-1',
    fileName: 'synth1.wav',
    category: 'Synth',
    dateAdded: new Date(2021, 12, 28),
    tags: ['synth', 'drone', 'cinematic'],
    tier: 'FREE',
  },
]
