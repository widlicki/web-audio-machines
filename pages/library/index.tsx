import type { NextPage } from 'next'
import AudioCategoryList from '../../components/audio-category-list/audio-category-list'
import WithRequest from '../../hoc/with-request'
import { useGetData } from '../../hooks/use-get-data'

const AudioCategoriesWithRequest = WithRequest(AudioCategoryList)

const LibraryPage: NextPage = () => {
  const {
    data: audioCategories,
    isLoading: isAudioCategoriesLoading,
    isError: isAudioCategoriesError,
  } = useGetData()

  return (
    <div>
      <h1>Library</h1>
      <AudioCategoriesWithRequest
        categories={audioCategories}
        isLoading={isAudioCategoriesLoading}
        isError={isAudioCategoriesError}
      />
    </div>
  )
}

export default LibraryPage
