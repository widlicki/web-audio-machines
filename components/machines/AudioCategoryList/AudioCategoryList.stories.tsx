import { ComponentMeta, ComponentStory } from '@storybook/react'
import { mockAudioCategories } from '../../../mocks/library'
import AudioCategoryList from './AudioCategoryList'

export default {
  title: 'Machines/AudioCategoryList',
  component: AudioCategoryList,
} as ComponentMeta<typeof AudioCategoryList>

const Template: ComponentStory<typeof AudioCategoryList> = (args) => (
  <AudioCategoryList {...args} />
)

export const WithData = Template.bind({})
WithData.args = {
  categories: mockAudioCategories,
}

export const WithoutData = Template.bind({})
WithoutData.args = {
  categories: [],
}
