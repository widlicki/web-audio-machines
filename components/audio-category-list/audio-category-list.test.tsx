import { render } from '@testing-library/react'
import { mockAudioCategories } from '../../mocks/library'
import AudioCategoryList from './audio-category-list'

let component: any

describe('AudioCategoryList', () => {
  beforeEach(() => {
    component = <AudioCategoryList categories={mockAudioCategories} />
  })

  it.skip('Should render AudioCategoryList according to snapshot', () => {
    const { container } = render(component)
    expect(container).toMatchSnapshot({ id: expect.any(String) })
  })

  it('Should render correct categories', () => {
    const { getByText } = render(component)

    mockAudioCategories.forEach((cat) => {
      expect(getByText(cat.name)).toBeTruthy()
    })
  })
})
