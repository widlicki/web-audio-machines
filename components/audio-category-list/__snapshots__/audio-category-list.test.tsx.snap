// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioCategoryList Should render AudioCategoryList according to snapshot 1`] = `
<div>
  <div
    class="css-0"
    style="visibility: visible; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-sjt5nk"
      id="menu-list-6039381599-2"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); visibility: visible; opacity: 1; transform: none;"
      tabindex="-1"
    >
      <button
        class="chakra-menu__menuitem css-o182o0"
        data-index="0"
        id="menu-list-6039381599-2-menuitem-6039381599-3"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Kick
      </button>
      <button
        class="chakra-menu__menuitem css-o182o0"
        data-index="1"
        id="menu-list-6039381599-2-menuitem-6039381599-4"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Bass
      </button>
      <button
        class="chakra-menu__menuitem css-o182o0"
        data-index="2"
        id="menu-list-6039381599-2-menuitem-6039381599-5"
        role="menuitem"
        tabindex="-1"
        type="button"
      >
        Synth
      </button>
    </div>
  </div>
</div>
`;
